### ISSUE TITLE
BarCodeScanner only scans on first mount Android

### ISSUE THREAD

**Issue by ajhopen** [expo/expo#15145]
### Summary

Responsive to upgrade from Expo SDK 42 => 43 on Android (only).  expo-barcode-scanner will only scan once.  After that, the camera will still come up but it will not detect any bar code.  To scan again you must not only close the application but also Expo Go and restart.  Then you will get one scan only before doing another restart.  

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

43

### Environment

 Expo CLI 4.13.0 environment info:
    System:
      OS: Windows 10 10.0.19042
    Binaries:
      Node: 14.17.3 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 7.20.5 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ^43.0.2 => 43.0.2
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: 0.17.1 => 0.17.1
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

This can be reproduced in the Basic BarCodeScanner usage snack from the Expo website API reference: exp://exp.host/@snack/sdk.43.0.0-fiKiGK2vqE

1. Scan using the snack once.
2. Close the snack keeping Expo Go open.
3. Reload the snack from Expo Go.  Notice it will not find a barcode.
4. Close the snack AND Expo Go. 
5. Start Expo Go again and load the snack.  You will can again (just once). 

In other words, it requires relaunching the Expo Go.  This is the same behavior our own applications.

**Comment by insoo-Jang** [comment]
I also have the same problem

you have to downgrade 41.0.1

**Comment by byCedric** [comment]
Hi @ajhopen and @insoo-Jang! Can one of you create a Snack or repo and link it here? The Snack url you linked is a temporary session, not the actual Snack. I'd be happy to take a look.

**Comment by ajhopen** [comment]
No problem @byCedric ...here you go:  exp://exp.host/@ajhopen/1x-barcode-scan-issue-sdk43+ad_6AImvVp

**Comment by byCedric** [comment]
> No problem @byCedric:
> 
> exp://exp.host/@ajhopen/1x-barcode-scan-issue-sdk43+ad_6AImvVp
> 
> If you stay on the same screen you can repeatedly read barcodes in SDK 43 all day. The issue is when you leave that initial screen and then come back to scan again. Some permission may be lost between 42=>43. Hopefully an easy fix. 🙏

This is a temporary session url again, please copy the normal url. It has to start with `https://snack.expo.dev/...`

**Comment by ajhopen** [comment]
@byCedric, sorry!   This should be it:  https://snack.expo.dev/@ajhopen/1x-barcode-scan-issue-sdk43

BTW, if you flip between SDK43 and SDK42 on the snack interface you'll see the issue in 43 and proper functioning in 42.

**Comment by fabriziocucci** [comment]
Affected by this as well, it looks like an unfortunate regression preventing people (like me) to migrate on SDK43.

**Comment by deywersonp** [comment]
I was affected too, I had to go back to version 42 due to the need to move forward with the project...

**Comment by ajhopen** [comment]
I updated the [test snack](https://snack.expo.dev/@ajhopen/1x-barcode-scan-issue-sdk43) to display the permissions results in the console: 

`Pixel 4: Scanscreen requestPermissionsAsync Status: granted
Pixel 4: Scanscreen requestPermissionsAsync Expires: never
Pixel 4: Scanscreen requestPermissionsAsync granted: true
Pixel 4: Scanscreen requestPermissionsAsync canAskAgain: true`

Doesn't seem to be a permissions issue.

**Comment by Mihai-github** [comment]
Hi, I ran into the same issue having inconsistent scanning experience running the expo SDK43. I've also checked the permission and they are ok ``` status: 'granted', canAskAgain: true, granted: true, expires: 'never ``` so it seems like there might be a different problem we are facing... 
Maybe someone has an idea or something to solve this problem?

**Comment by ajhopen** [comment]
@Mihai-github, maybe start looking at other properties like BarCodeBounds, BarCodePoint, etc... and see if anything is different between SDK42 and SDK43.  Whatever is happening requires a reboot...very mysterious.  @byCedric any suggestions to escalate this?

**Comment by byCedric** [comment]
> @Mihai-github, maybe start looking at other properties like BarCodeBounds, BarCodePoint, etc... and see if anything is different between SDK42 and SDK43. Whatever is happening requires a restart...very mysterious. @byCedric any suggestions to escalate this?

In my testing from Friday, it seems that this bug is specific to Expo Go. Building the app locally through Android Studio (after running `expo prebuild`) seems to work as expected. You could try building your app with EAS, [it has a free tier now](https://twitter.com/expo/status/1458598050310475776), and see if you have the same behavior.

For now, I'll mark it as a bug, but it's still unclear where the issue is coming from.

**Comment by Mihai-github** [comment]
@byCedric HI, i was able in local environment to make it work by changing the version of expo, from 43 to 42.
This issue seems to happen also if you go try the expo snack for barcode provided in the docs having the version of expo sdk set by default to 43 after first scanning something happends internally or idk that makes the camera used by expo-barcode-scanner not scanning the codes anymore.

You can try the snack that is available in docs and see what i am talking about.
Maybe this will help the dev team to fix this. 

Thanks 😊, Let me know if something changes please 😀

**Comment by eduayrescarneiro** [comment]
i am also having the same problem.

**Comment by ajhopen** [comment]
@byCedric, agreed, looks like an Android Expo Go issue.  I'm on Version 2.22.3 updated on Oct. 27, 2021.  At least it is 100% reproducible. Anything else we can do to advance a resolution?

**Comment by byCedric** [comment]
Hi @ajhopen! I'm trying to get the right people on this internally asap. Hope to give you good news soon!

Meanwhile, the current workaround is using EAS (which has a free tier) to build the app like a bare app, even when using a managed app. There are more benefits like you aren't tied to actually using EAS; you can also just build it locally with `eas build --local`.

**Comment by eduayrescarneiro** [comment]
@ajhopen I changed my implementation and used the Camera component with the onBarCodeScanned prop. It worked like a charm!

**Comment by edi** [comment]
Does the barcode scanner (via `expo-camera`) work for you at all on iOS guys?
For me, it literally does not scan anything at all. And I can't use `expo-barcode-scanner`, because it has no `torch` controls.
Please advise

**Comment by ajhopen** [comment]
> @ajhopen I changed my implementation and used the Camera component with the onBarCodeScanned prop. It worked like a charm!

@eduayrescarneiro ... Brilliant!  All I had to do:

`import { Camera } from 'expo-camera';
// change below
  //const { status } = await BarCodeScanner.requestPermissionsAsync();
      const { status } = await Camera.requestCameraPermissionsAsync();
`
and change           <BarCodeScanner...> to  <Camera...>

Works great on Android Expo Go.

**Comment by ayez0001** [comment]
@eduayrescarneiro, @ ajhopen   your solution works well. Thank you  :)

**Comment by barbhackk** [comment]
Hi,

This seems like a good solution but we cannot retrieve the bounding box and other values with Camera. Not ?

Thanks

**Comment by barbhackk** [comment]
Hi all,

Try this solutions : https://github.com/expo/expo/issues/9619#issuecomment-980723755

It's work for me but not the best practice. This allows to unblock while waiting for a correction.

**Comment by wangtiger317** [comment]
> > @ajhopen I changed my implementation and used the Camera component with the onBarCodeScanned prop. It worked like a charm!
> 
> @eduayrescarneiro ... Brilliant! Well done sir. Following your suggestions, all I had to do:
> 
> import { **Camera** } from '**expo-camera**'; // change below //const { status } = await BarCodeScanner.requestPermissionsAsync(); const { status } = await **Camera.requestCameraPermissionsAsync**();
> 
> and change <BarCodeScanner...> to <**Camera**...>
> 
> Works great on Android Expo Go.

for the barcode , it doesnt work, only it is working for the qr code scanner
